x-pack/plugins/fleet/server/services/epm/packages/get.ts

```diff
@@ -169,6 +170,45 @@ export const getPackageUsageStats = async ({
   };
 };
 
+export async function getPackagePermissions(
+  soClient: SavedObjectsClientContract,
+  pkgName: string,
+  pkgVersion: string,
+  namespace = '*'
+): Promise<PackagePermissions | undefined> {
+  const pkg = await getPackageInfo({ savedObjectsClient: soClient, pkgName, pkgVersion });
+  if (!pkg.data_streams) {
+    return undefined;
+  }
+
+  const clusterPermissions = new Set<string>();
+  const indices: PackagePermissions['indices'] = pkg.data_streams!.map((ds) => {
+    if (ds.permissions?.cluster) {
+      ds.permissions.cluster.forEach((p) => clusterPermissions.add(p));
+    } else {
+      clusterPermissions.add('monitor');
```

Comment:
I wonder if we should throw `monitor` into its own "entry" of permissions so we don't have to deal with it in here?

Reply:
I'm assuming by its own entry you mean an extra key-value in the agent policy

```yaml
output_permissions:
  default:
    some_new_entry:
      cluster:
        - monitor
    system-1:
      indices:
        - names:
            - logs-system.application-test
          privileges:
            - auto_configure
            - create_doc
        - names:
            - logs-system.auth-test
          privileges:
            - auto_configure
            - create_doc
        # ...
```

Imagine a package with two datasets. I see three use cases.

1. No dataset specifices `cluster` permissions. The result should be `cluster: ['monitor']`
2. One of the dataset specifies some `cluster` permissions (let's call it `foo`). The result should be `cluster: ['monitor', 'foo']`.
3. Both datasets specify cluster permissions (let's call them `foo` for dataset1, and `bar` for dataset2). The result should be `cluster: ['foo', 'bar']`.

I'm not sure how to handle case 3 if the `monitor` permission is its own entry. If the policy only has one package/integration and none of its datasets specify that permission we will add it for no reason. That is, unless the `monitor` permission is added for some other reason I'm not aware of.

Reply:
AFAIK the monitoring permission is required at the moment because of the Elastic Agent license check and is not related to any integration. Maybe we should even call the block `elastic-agent-checks` or similar to make it clear it is not related to all the other parts.

I'm hoping non of the packages will require cluster permissions but I'm pretty sure we can find an example to proof me wrong.

Reply:
I changed it 👍